### PR TITLE
feat: implement `@sit-onyx/playwright-utils` package

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to #1991

Implement `useMatrixScreenshotTest` and `adjustSizeToAbsolutePosition` playwright utilities.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/playwright-utils/src/screenshots/utils.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Escapes the given string so its save to be used as CSS grid-area, e.g. by replacing whitespaces.
+ */
+export const escapeGridAreaName = (name: string) => {
+  return name.replace(/ /g, "-");
```
Comment (JoCa96) [inline packages/playwright-utils/src/screenshots/utils.ts:5]: To be a bit pedantic here 😅: 
We should probably replace all unusual characters:
```suggestion
  return name.replace(/\W/g, "-");
```

File: packages/playwright-utils/src/screenshots/types.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+   * **IMPORTANT**: Should be avoided! If used, please include a comment why it is needed.
+   *
+   * @see https://playwright.dev/docs/accessibility-testing#disabling-individual-scan-rules
+   */
+  disabledAccessibilityRules?: string[];
+  /**
+   * If `true`, no padding will be added to the screenshots.
+   * By default a `1rem` padding is added around the component/screenshot.
+   */
+  disablePadding?: boolean;
```
Comment (JoCa96) [inline packages/playwright-utils/src/screenshots/types.ts:74]:
```suggestion
  removePadding?: boolean;
```

File: packages/playwright-utils/src/screenshots/types.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  /**
+   * Function that returns the component for the given column and row.
+   */
+  component: (column: TColumn, row: TRow) => JSX.Element;
+  /**
+   * Optional callback to be executed before capturing the screenshot.
+   * Useful for performing `expect()` or e.g. hover, focus-visible state etc.
+   * Focus and mouse will be reset after each screenshot.
+   */
+  beforeScreenshot?: ScreenshotTestHook<TColumn, TRow>;
```
Comment (JoCa96) [inline packages/playwright-utils/src/screenshots/types.ts:56]: Would prefer to rename or clarify in the description that this is called before an **individual/single** screenshot is taken
```suggestion
  beforeScreenshot?: ScreenshotTestHook<TColumn, TRow>;
```

File: packages/playwright-utils/src/screenshots/index.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+        if (disabledAccessibilityRules.length) {
+          axeBuilder.disableRules(disabledAccessibilityRules);
+        }
+
+        const accessibilityScanResults = await axeBuilder.analyze();
+        expect(
+          accessibilityScanResults.violations,
+          `should pass accessibility checks for ${column} ${row}`,
+        ).toEqual([]);
```
Comment (JoCa96) [inline packages/playwright-utils/src/screenshots/index.tsx:57]: Imho we should remove axe-core as a default dependency/include as it addresses a different concern and could be called as well in the `afterScreenshot` hook.

File: packages/playwright-utils/src/screenshots/index.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+        const id = `${row}-${column}`;
+
+        const image = (
+          <img
+            width={box?.width}
+            height={box?.height}
+            style={{ gridArea: escapeGridAreaName(id) }}
+            src={`data:image/png;base64,${Buffer.from(screenshot).toString("base64")}`}
```
Comment (JoCa96) [inline packages/playwright-utils/src/screenshots/index.tsx:66]: Please avoid base64 conversion as it unnecessarily increases memory and process usages. I thought we already, addressed this? 🤔

**Review by JoCa96 — CHANGES_REQUESTED** [review]
Please move axe-core checks out

File: packages/playwright-utils/src/screenshots/index.tsx (hunk 4 above)
Comment (larsrickert) [reply packages/playwright-utils/src/screenshots/index.tsx]: I refactored the utility with a new `hooks` pattern, what do you think?

Comment (JoCa96) [reply packages/playwright-utils/src/screenshots/index.tsx]: Looks great! Amazing work :)

File: packages/playwright-utils/src/screenshots/index.tsx (hunk 5 above)
Comment (larsrickert) [reply packages/playwright-utils/src/screenshots/index.tsx]: Done. However, after local testing it seems to me that base64 conversion is actually faster than mocking the Playwright routes and doing actual requests.

I will refine a follow up ticket for this to check it: #2291

**Review by JoCa96 — APPROVED** [review]

**Pull request merged**